### DISCUSSION TITLE
Dynamic Page with persisted Redux

### DISCUSSION THREAD

**Discussion by vladfulgeanu** [vercel/next.js#11175] [Help]
Thanks for an amazing library!
I'm new to next.js so I'm looking to start a discussion for what's the best way to solve this:

I'm using persisted redux using `next-redux-wrapper` with localStorage and until now most components connected to redux with `mapStateToProps` and selected data using `reselect`.

What I have now is a dynamic article page which should be SEO friendly but which can also show client specific info if the user is logged in like _save article_ / _article already saved_.

1. How can I properly return the page info using `getServerSideProps` when landing on the page, but when using transitions from other pages use the data that is already available in redux store (get articleId from query and use a selector for getting data and pass it to props with `mapStateToProps`) and not let next.js make a new request like it is mentioned in the [docs](https://nextjs.org/docs/basic-features/data-fetching#only-runs-on-server-side)?
1.1 Also if the user has already logged in and the data is persisted in localStorage, how can I get that data instead of making a new request on server side?
2. If I have thousands of article pages and also if I want to create client side features (_save article_) is it feasible to use SSG (getStaticProps and getStaticPaths)?

**Comment by willemliu** [comment]
I don't think I quite understand what `save article` does in your example. Do you mean you have article pages which can be edited similar to a wikipedia page?

**Reply by vladfulgeanu** [reply]
`save article` will save them to your profile for reading later, you can think of it as liking an article, and when you're logged in it should be shown in an article's page that you've liked that article.
